examples/avro-cli.py

```diff
@@ -0,0 +1,194 @@
+#!/usr/bin/env python
+#
+# Copyright 2018 Confluent Inc.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+# http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+
+import argparse
+from uuid import uuid4
+
+from six.moves import input
+
+from confluent_kafka import avro
+
+# Parse Schema used for serializing User class
+record_schema = avro.loads("""
+    {
+        "namespace": "confluent.io.examples.serialization.avro",
+        "name": "User",
+        "type": "record",
+        "fields": [
+            {"name": "name", "type": "string"},
+            {"name": "favorite_number", "type": "int"},
+            {"name": "favorite_color", "type": "string"}
+        ]
+    }
+""")
+
+
+class User(object):
+    """
+        User stores the deserialized user Avro record.
+    """
+
+    # Use __slots__ to explicitly declare all data members.
+    __slots__ = ["name", "favorite_number", "favorite_color", "id"]
+
+    def __init__(self, name=None, favorite_number=None, favorite_color=None):
+        self.name = name
+        self.favorite_number = favorite_number
+        self.favorite_color = favorite_color
+        # Unique id used to track produce request success/failures.
+        # Do *not* include in the serialized object.
+        self.id = uuid4()
+
+    def to_dict(self):
+        """
+            The Avro Python library does not support code generation.
+            For this reason we must provide a dict representation of our class for serialization.
+        """
+        return {
+            "name": self.name,
+            "favorite_number": self.favorite_number,
+            "favorite_color": self.favorite_color
+        }
+
+
+def on_delivery(err, msg, obj):
+    """
+        Handle delivery reports served from producer.poll.
+        This callback takes an extra argument, obj.
+        This allows the original contents to be included for debugging purposes.
+    """
+    if err is not None:
+        print('Message {} delivery failed for user {} with error {}'.format(
+                                                                            obj.id,
```

Comment:
micro nit:
this indent-style is kind of weird, continuations are typically current_indent + 1 tab, otherwise
there isn't much horizontal space saved by the continuation.

I guess this is a vim thing?